libs/common/src/key-management/crypto/services/encrypt.service.implementation.ts

```diff
@@ -211,42 +211,22 @@ export class EncryptServiceImplementation implements EncryptService {
     return result ?? null;
   }
 
-  async rsaEncrypt(data: Uint8Array, publicKey: Uint8Array): Promise<EncString> {
-    if (data == null) {
-      throw new Error("No data provided for encryption.");
-    }
-
-    if (publicKey == null) {
-      throw new Error("No public key provided for encryption.");
+  async encapsulateKeyUnsigned(key: SymmetricCryptoKey, publicKey: Uint8Array): Promise<EncString> {
+    if (key == null) {
+      throw new Error("No key");
     }
-    const encrypted = await this.cryptoFunctionService.rsaEncrypt(data, publicKey, "sha1");
-    return new EncString(EncryptionType.Rsa2048_OaepSha1_B64, Utils.fromBufferToB64(encrypted));
+    const keyBytes = key.key;
+    const encryptedKey = await this.rsaEncrypt(keyBytes, publicKey);
+    return encryptedKey;
   }
 
-  async rsaDecrypt(data: EncString, privateKey: Uint8Array): Promise<Uint8Array> {
-    if (data == null) {
-      throw new Error("[Encrypt service] rsaDecrypt: No data provided for decryption.");
-    }
-
-    let algorithm: "sha1" | "sha256";
-    switch (data.encryptionType) {
-      case EncryptionType.Rsa2048_OaepSha1_B64:
-      case EncryptionType.Rsa2048_OaepSha1_HmacSha256_B64:
-        algorithm = "sha1";
-        break;
-      case EncryptionType.Rsa2048_OaepSha256_B64:
-      case EncryptionType.Rsa2048_OaepSha256_HmacSha256_B64:
-        algorithm = "sha256";
-        break;
-      default:
-        throw new Error("Invalid encryption type.");
-    }
-
-    if (privateKey == null) {
-      throw new Error("[Encrypt service] rsaDecrypt: No private key provided for decryption.");
-    }
-
-    return this.cryptoFunctionService.rsaDecrypt(data.dataBytes, privateKey, algorithm);
+  async decapsulateKeyUnsigned(
+    data: EncString,
+    privateKey: Uint8Array,
+  ): Promise<SymmetricCryptoKey> {
+    const keyBytes = await this.rsaDecrypt(data, privateKey);
+    const key = new SymmetricCryptoKey(keyBytes);
+    return key;
```

Comment:
```suggestion
  async decapsulateKeyUnsigned(
    encapsulatedKey: EncString,
    decapsulationKey: Uint8Array,
  ): Promise<SymmetricCryptoKey> {
    const keyBytes = await this.rsaDecrypt(encapsulatedKey, decapsulationKey);
    return new SymmetricCryptoKey(keyBytes);
```

Matching abstraction parameter names. ⛏️  Having the `const key` doesn't help with readability IMO.

Reply:
Changing to "sharedKey", "encapsulationKey", and "decapsulationKey" (https://github.com/bitwarden/sdk-internal/pull/202#discussion_r2026869399)